### ISSUE TITLE
A value is trying to be set on a copy of a slice from a DataFrame.

### ISSUE THREAD

**Issue by blablax** [whittlem/pycryptobot#182]
Hi,

updated to 2.1.0 and this message is displayed:
A value is trying to be set on a copy of a slice from a DataFrame.

|         Iteration : 1 (BEAR)                                                 |
|         Timestamp : 2021-05-14 08:00:00                                      |
--------------------------------------------------------------------------------
|             Close : 41649.67                                                 |
|             EMA12 : 41113.39                                                 |
|             EMA26 : 41443.94                                                 |
|    Crossing Above : False                                                    |
|   Currently Above : False                                                    |
|    Crossing Below : False                                                    |
|   Currently Below : True                                                     |
|         Condition : EMA12 is currently below EMA26 and has crossed over      |
|             SMA20 : 40947.21                                                 |
|            SMA200 : 46134.6                                                  |
--------------------------------------------------------------------------------
|              MACD : -330.55                                                  |
|            Signal : -542.7                                                   |
|   Currently Above : True                                                     |
|   Currently Below : False                                                    |
|         Condition : MACD is currently above Signal and has crossed over      |
--------------------------------------------------------------------------------
|            Action : WAIT                                                     |
================================================================================
/opt/pycrypt/bot-btc-eur/models/exchange/coinbase_pro/api.py:214: SettingWithCopyWarning:
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versu
s-a-copy
  df['price'] = df.apply(lambda row: round((float(row.executed_value) * 100) / (float(row.filled_size) * 100), 2) if float(row.fil
led_size) > 0 else 0, axis=1)
^C2021-05-14 08:59:29.426749 closed

Config:

    "coinbasepro" : {
        "api_url" : "https://api.pro.coinbase.com",
        "api_key" : "",
        "api_secret" : "",
        "api_passphrase" : "",
        "config" : {
            "base_currency" : "BTC",
            "quote_currency" : "EUR",
            "granularity" : "120",
            "live" : 1,
            "verbose" : 1,
            "graphs": 1,
            "graph": 1,
            "save_graphs": 1
        }
    },
    "telegram" : {
    "token" : ":",
    "client_id" : ""
    }
}

Are there more people experiencing this?

**Comment by cr0sis** [comment]
Yeah, I'm seeing the same

**Comment by HeadSol** [comment]
Same issue here, as well...

```
{

    "coinbasepro" : {
        "api_url" : "https://api.pro.coinbase.com",
        "api_key" : "",
        "api_secret" : "",
        "api_passphrase" : "",
        "config" : {
            "granularity" : "120",
            "base_currency" : "ETC",
            "quote_currency" : "USD",
            "live" : 1,
            "sellatloss" : 1,
            "disablebullonly" : 1,
            "disablebuyobv" : 1,
            "disablebuyelderray" : 1,
            "disablelog" : 1,
            "verbose" : 1
        }
    }

}
```

**Comment by whittlem** [comment]
I've never seen this myself. Are you still getting this error on the latest version?

It looks like a Pandas warning rather than an error. Maybe you are running a different version of Pandas to me?

% python3 -m pip freeze | grep pandas
pandas==1.1.4

Can you confirm which version of Pandas you are using and if it is different to mine please install the version I have and let me know if this resolves your issue?

**Comment by Nicolas469** [comment]
I've also had this error.
I'm running the bot in Docker so the pandas version is the last one (1.2.4) since requirements.txt don't set the version.
I guess that the simple solution is to put the version (1.1.4) in requirements.txt.

**Comment by Askeo** [comment]
had the same error, fresh installed with fresh requirements and with fresh directory, 2 folders 1 BTC-EUR 1 ADA-EUR

ADA-EUR is no error 
BTC-EUR has the same error dont know why its almost the same folder

tested with MATIC-EUR aswell error

**Comment by rwl816** [comment]
I'm also having the same issue.   I get the error when i set it up for BTC-USD, but when its set up for UMA-USD there is no error.  it seems to be happening in the getOrders def in api.py when its converting the values in the DataFrame df.  
i've installed v2.23.3 and confirmed that pandas is version 1.1.4
any ideas?

**Comment by whittlem** [comment]
This is fixed in v2.25.0
